server/src/main/java/org/elasticsearch/action/admin/cluster/node/info/TransportNodesInfoAction.java

```diff
@@ -76,41 +78,72 @@ protected NodeInfo newNodeResponse(StreamInput in, DiscoveryNode node) throws IO
 
     @Override
     protected NodeInfo nodeOperation(NodeInfoRequest nodeRequest, Task task) {
-        NodesInfoRequest request = nodeRequest.request;
-        Set<String> metrics = request.requestedMetrics();
+        Set<String> metrics = nodeRequest.requestedMetrics();
         return nodeService.info(
-            metrics.contains(NodesInfoRequest.Metric.SETTINGS.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.OS.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.PROCESS.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.JVM.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.THREAD_POOL.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.TRANSPORT.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.HTTP.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.REMOTE_CLUSTER_SERVER.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.PLUGINS.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.INGEST.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.AGGREGATIONS.metricName()),
-            metrics.contains(NodesInfoRequest.Metric.INDICES.metricName())
+            metrics.contains(NodesInfoMetrics.Metric.SETTINGS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.OS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.PROCESS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.JVM.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.THREAD_POOL.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.TRANSPORT.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.HTTP.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.REMOTE_CLUSTER_SERVER.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.PLUGINS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.INGEST.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.AGGREGATIONS.metricName()),
+            metrics.contains(NodesInfoMetrics.Metric.INDICES.metricName())
         );
     }
 
     public static class NodeInfoRequest extends TransportRequest {
```

Comment:
The changes to `NodeInfoRequest` seem unrelated to the rest of this PR, would you revert them?

Reply:
Thanks. The changes to `NodeInfoRequest` is to decouple it from `TransportNodesInfoAction.NodeInfoRequest`. Then we can trim down the metrics that `NodeInfoRequest` carries. Do you mean we just extract `NodeInfoMetrics` to a top-level class from NodesInfoRequest and then do that changes in the next PR?

Reply:
Right, let's _only_ extract the metrics to a top-level class first, then we can work on trimming down `NodeInfoRequest` in a follow-up.

Reply:
done